File: Descomposición factores primos/Python/descomposicion_factores_primos_v2_1.py
#Creado por Andrés Melgar
#para iterando++ http://iterando.online/
#entrada: http://iterando.online/ejercicios-resueltos/descomposicion-en-factores-primos-en-python/
import math

def es_primo(n):
    if n <= 0:
        return False
    cant_divisores = 0
    encontro_divisores = False
    limite=math.sqrt(n)
    i = 2
    while (i <= limite and not encontro_divisores):
        if n % i == 0:
            cant_divisores+=1
            encontro_divisores = True
        i+=1
    if cant_divisores==0 and n>1:
        return True
    return False
        
def siguiente_primo(n):
    while True:
        n+=1
        if es_primo(n):
            return n
    
def descomponer_factores(n):
    productoria = 1
    copia_n = n
    factor_primo = 2
    cant_factores = 0
    hay_factores = False
    while True:
        resto = copia_n % factor_primo
        if resto == 0:            
            copia_n //= factor_primo
            productoria *= factor_primo
            cant_factores += 1
        else:  
            if cant_factores > 0:
                if hay_factores:
                    print("x", end="")
                else:
                    hay_factores = True
                if cant_factores==1:
                    print(factor_primo, end="")
                else:
                    print("{:d}^{:d}".format(factor_primo, cant_factores), end="")
                cant_factores = 0
            factor_primo = siguiente_primo(factor_primo)
        
        if productoria == n:
            break;
    if hay_factores:
        print("x", end="")
    if cant_factores==1:
        print(factor_primo, end="")
    else:
        print("{:d}^{:d}".format(factor_primo, cant_factores), end="")
    
n=int(input("Ingrese un número (>1): "))
if n <= 1:
    print("Debe ingresar un número mayor que uno")
else:
    descomponer_factores(n)

File: Descomposición factores primos/Java/Descomposicion_factores_primos_v2_1.java
/* Creado por Andrés Melgar
 * para iterando++ http://iterando.online/
 * entrada: http://iterando.online/ejercicios-resueltos/descomposicion-en-factores-primos-en-java/
 */
package descomposicion_factores_primos_v2_1;

import static java.lang.Math.sqrt;
import java.util.Scanner;

public class Descomposicion_factores_primos_v2_1 {

    public static void main(String[] args) {
        int n;
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese un número (>1): ");
        n = reader.nextInt();

        if (n <= 1) {
            System.out.printf("Debe ingresar un número mayor que uno%n");
        } else {
            descomponer_factores(n);
        }
    }

    private static void descomponer_factores(int n) {
        int productoria = 1;
        int copia_n = n;
        int factor_primo = 2;
        int cant_factores = 0;
        boolean hay_factores = false;
        do {
            int resto = copia_n % factor_primo;
            if (resto == 0) {
                copia_n /= factor_primo;
                productoria *= factor_primo;
                cant_factores++;
            } else {
                if (cant_factores > 0) {
                    if (hay_factores) {
                        System.out.printf("x");
                    } else {
                        hay_factores = true;
                    }
                    if (cant_factores == 1) {
                        System.out.printf("%d", factor_primo);
                    } else {
                        System.out.printf("%d^%d", factor_primo, cant_factores);
                    }
                    cant_factores = 0;
                }
                factor_primo = siguiente_primo(factor_primo);
            }
        } while (productoria != n);
        if (hay_factores) {
            System.out.printf("x");
        }
        if (cant_factores == 1) {
            System.out.printf("%d", factor_primo);
        } else {
            System.out.printf("%d^%d", factor_primo, cant_factores);
        }
    }

    private static int siguiente_primo(int n) {
        do {
            n++;
        } while (!es_primo(n));
        return n;
    }

    private static boolean es_primo(int n) {
        if (n <= 0) {
            return false;
        }
        int cant_divisores = 0;
        boolean encontro_divisores = false;
        int limite = (int) sqrt(n);
        int i = 2;
        while (i <= limite && !encontro_divisores) {
            if (n % i == 0) {
                cant_divisores++;
                encontro_divisores = true;
            }
            i++;
        }
        if (cant_divisores > 0 || n == 1) {
            return false;
        }
        return true;
    }
}